File: benchmark/http/go-fiber/internal/wrkload/handler.go
package wrkload

import (
	"benchmark-go-http/pkg/utils"
	"strconv"

	"github.com/gofiber/fiber/v3"
)

type WrkloadHandler struct {
	service WrkloadService
}

func NewWrkloadHandler(svc WrkloadService) *WrkloadHandler {
	return &WrkloadHandler{
		service: svc,
	}
}

func (h *WrkloadHandler) RegisterRoutes(app *fiber.App) {
	app.Get("/wrk", h.Wrk)
}


func (h *WrkloadHandler) Wrk(c fiber.Ctx) error {
	endpoint := c.Query("e", "https://example.com")
	method := c.Query("m", "get")

	connectionsStr := c.Query("c", "10")
	durationStr := c.Query("d", "10")

	connections, err := strconv.Atoi(connectionsStr)
	if err != nil {
		return c.Status(fiber.StatusBadRequest).JSON(fiber.Map{
			"error": "invalid connections value",
		})
	}

	duration, err := strconv.Atoi(durationStr)
	if err != nil {
		return c.Status(fiber.StatusBadRequest).JSON(fiber.Map{
			"error": "invalid duration value",
		})
	}

	apiDataB := ApiDataBuilder{}
	apiData := apiDataB.SetUrl(endpoint).SetMethod(method).Build()

	wrkResult, err := h.service.Wrk(c, apiData, uint(connections), uint(duration))

	if err != nil {
		return utils.ErrorResponse(
			c, fiber.StatusInternalServerError, err.Error(),
		)
	}
	return utils.SuccessResponse(c, &WrkResponse{
		Url: wrkResult.Url,
		Method: wrkResult.Method,
		TimeTaken: wrkResult.TimeTaken,
		Connections: wrkResult.Connections,
		TotalHits: wrkResult.TotalHits,
		SuccessHits: wrkResult.SuccessHits,
		FailureHits: wrkResult.FailureHits,
		SuccessMessages: wrkResult.SuccessMessages,
		FailureMessages: wrkResult.FailureMessages,
	})
}

File: benchmark/http/fastapi/app/handler.py
from fastapi import APIRouter, Query, HTTPException
from models import ApiData
from service import WrkloadService

router = APIRouter()
service = WrkloadService()


@router.get("/wrk")
async def wrk(
    e: str = Query("https://example.com"),
    m: str = Query("get"),
    c: int = Query(10),
    d: int = Query(10),
):
    if c <= 0:
        raise HTTPException(status_code=400, detail="invalid connections value")

    if d <= 0:
        raise HTTPException(status_code=400, detail="invalid duration value")

    api_data = ApiData(url=e, method=m)

    result = await service.wrk(
        api_data=api_data,
        connections=c,
        duration=d
    )

    return result